src/array.jl

```diff
@@ -69,6 +87,30 @@ Base.print_array(io::IO, X::LinearAlgebra.AdjOrTrans{Bool, <:OneHotLike{T, L, N,
 _onehot_bool_type(::OneHotLike{<:Any, <:Any, <:Any, N, <:Union{Integer, AbstractArray}}) where N = Array{Bool, N}
 _onehot_bool_type(::OneHotLike{<:Any, <:Any, <:Any, N, <:CuArray}) where N = CuArray{Bool, N}
 
+_onehot_compatible(x::OneHotLike) = _isonehot(x)
+_onehot_compatible(x::AbstractVector{Bool}) = count(x) == 1
+_onehot_compatible(x::AbstractArray{Bool}) = all(isone, reduce(+, x; dims=1))
```

Comment:
Slightly tidier, maybe:
```suggestion
_onehot_compatible(x::AbstractArray{Bool}) = all(isone, count(x; dims=1))
```